src/app/pages/OpenWalletPage/__tests__/index.test.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { render, screen, waitFor } from '@testing-library/react'
+import { MemoryRouter } from 'react-router-dom'
+import { useSelector } from 'react-redux'
+import { when } from 'jest-when'
+import { selectShowAccountsSelectionModal } from 'app/state/importaccounts/selectors'
+import { canAccessNavigatorUsb } from 'app/lib/ledger'
+import { SelectOpenMethod } from '..'
+
+jest.mock('react-redux', () => ({
+  useSelector: jest.fn(),
+}))
+jest.mock('app/lib/ledger')
+
+const mockNavigate = jest.fn()
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useNavigate: () => mockNavigate,
+}))
+
+const renderComponent = (webExtensionLedgerAccess?: () => void) =>
+  render(
+    <MemoryRouter>
+      <SelectOpenMethod webExtensionLedgerAccess={webExtensionLedgerAccess} />
+    </MemoryRouter>,
+  )
+
+describe('<SelectOpenMethod />', () => {
+  beforeEach(() => {
+    jest.mocked(canAccessNavigatorUsb).mockResolvedValue(false)
+  })
+
+  it('should render component', async () => {
+    const { container } = await waitFor(() => renderComponent())
+
+    expect(container).toMatchSnapshot()
+  })
+
+  it('should render component with an access button', async () => {
+    await waitFor(() => renderComponent(() => {}))
```

Comment:
??

waitFor runs the function multiple times until expect() inside is successful. So this renders infinite number of components, or I guess it does nothing because there are no expects inside